### PR TITLE
Modify rule S1862 (`no-identical-conditions`): Consider identical cases of switch statements

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes https://github.com/SonarSource/SonarJS/issues/3700

### CODE REVIEW

File: docs/rules/no-identical-conditions.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # no-identical-conditions
 
-A chain of `if`/`else if` statements is evaluated from top to bottom. At most, only
-one branch will be executed: the first one with a condition that evaluates to `true`.
+A chain of `if-else-if` and `switch-case` statements is evaluated from top to bottom. At most, only
+one branch will be executed: the first one with a condition that evaluates to `true` or that matches the discriminant.
```
Comment (ilia-kebets-sonarsource) [inline docs/rules/no-identical-conditions.md:4]:
```suggestion
one branch will be executed: the first one with a condition that evaluates to `true` (or that matches the discriminant for the switch-case).
```

File: src/rules/no-identical-conditions.ts [hunk 2]
```diff
@@ -77,6 +81,34 @@ const rule: TSESLint.RuleModule<string, string[]> = {
...
+                  data: {
+                    line: duplicateTest.loc.start.line,
+                  },
+                  node: test,
+                },
+                [issueLocation(duplicateTest.loc, duplicateTest.loc, 'Original')],
+                duplicatedCaseMessage,
+              );
+            } else {
+              previousTests.push(test);
```
Comment (ilia-kebets-sonarsource) [inline src/rules/no-identical-conditions.ts:107]: What about adding a test case that shows that when you have more than 1 duplicate, they both point to the same first "duplicate"?

**Review by ilia-kebets-sonarsource — APPROVED** [review]
See comments otherwise LGTM!
